libp2p/multistream.nim

```diff
@@ -261,20 +261,22 @@ proc addHandler*(m: MultistreamSelect,
                                match: matcher))
 
 proc start*(m: MultistreamSelect) {.async: (raises: [CancelledError]).} =
-  let
-    handlers = m.handlers
-    futs = handlers.mapIt(it.protocol.start())
+  # Nim 1.6.18: Using `mapIt` results in a seq of `.Raising([])`
```

Comment:
er .. that doesn't make sense at all - should result in `.Raising([CancelledError])` since that's what start does now..
```nim
import chronos, sequtils

proc x(v: int) {.async: (raises: [CancelledError]).} = discard

var s = mapIt([1, 2, 3], x(it))

static: echo type(s)
```
nim prints:
```
seq[InternalRaisesFuture[system.void, (CancelledError,)]]
```

Reply:
Also, interestingly, in `stop` there is the reverse problem where `mapIt` invents a random `.Raising([CancelledError])`

Reply:
```nim
import chronos, sequtils

proc start(v: int) {.async: (raises: [CancelledError]).} = discard
proc stop(v: int) {.async: (raises: []).} = discard

var a = mapIt([1, 2, 3], start(it))

static: echo type(a)
var b = mapIt([1, 2, 3], stop(it))

static: echo type(b)
```
funny bunny.

```
seq[InternalRaisesFuture[system.void, (CancelledError,)]]
seq[InternalRaisesFuture[system.void, (CancelledError,)]]
```

Reply:
oh, nice that you could find a more minimal version that's actually postable to nim-lang :) less painful than expected. have run into these separately, in different PRs. 

`stop` was fixed here: 

- https://github.com/vacp2p/nim-libp2p/pull/1066

While `start` gets fixed in the current PR.

Funny that they interfere with each other, didn't expect there to be a link between `start` and `stop`.

Could it be a name clash thing with `chronos`?

Reply:
https://github.com/nim-lang/Nim/issues/23418

Reply:
https://github.com/status-im/nim-chronos/pull/522